### PR TITLE
CORE-17447: Add subscriptions for cleanup processors in flow mapper

### PR DESCRIPTION
**Pull request by JamesHR3**
### Problem statement

The flow mapper cleanup work has components to process events present. These need to be plumbed into the flow mapper service.

### Solution

Plumb these components in.

This involves setting up durable subscriptions and defining processors for the corresponding events. The main issue is that integration tests involving the state manager are currently challenging to add in a way that allows them to run outside of postgres. This change sidesteps the problem for now by introducing a dummy component, but a better test would use a more realistic state manager implementation. This can be addressed in a follow up.

### CODE REVIEW

**Review by ronanbrowne — DISMISSED** [review]
LGTM (blt file only reviewed)

File: charts/corda/values.schema.json [hunk 1]
```diff
@@ -2022,6 +2023,66 @@
                                     "examples": [
                                         false
                                     ]
+                                },
+                                "stateManagerDbConnectionPool": {
+                                    "type": "object",
+                                    "default": {},
+                                    "title": "Flow worker JDBC connection pool configuration for State Manager DB",
```
Comment (davidcurrie) [inline charts/corda/values.schema.json]: Either the comments need updating to reference flow mapper or move the whole thing down into the `$defs` section, make the comments worker agnostic,  and reference it from both places.

**Review by davidcurrie — APPROVED** [review]
LGTM for chart changes

File: libs/sandbox-internal/src/integrationTest/kotlin/net/corda/sandboxtests/SandboxBundleDifferentLibrariesTest.kt [hunk 2]
```diff
@@ -29,15 +29,15 @@ class SandboxBundleDifferentLibrariesTest {
...
+        @InjectService(timeout = 10000)
         sandboxSetup: SandboxSetup,
         @InjectBundleContext
         bundleContext: BundleContext,
         @TempDir testDirectory: Path
     ) {
         sandboxSetup.configure(bundleContext, testDirectory)
         lifecycle.accept(sandboxSetup) { setup ->
-            sandboxFactory = setup.fetchService(timeout = 1000)
+            sandboxFactory = setup.fetchService(timeout = 10000)
```
Comment (jujoramos) [inline libs/sandbox-internal/src/integrationTest/kotlin/net/corda/sandboxtests/SandboxBundleDifferentLibrariesTest.kt:40]: Are these changes intentional?.

**Review by jujoramos — APPROVED** [review]

File: components/flow/flow-mapper-service/src/test/kotlin/net/corda/session/mapper/service/executor/ScheduledTaskProcessorTest.kt [hunk 3]
```diff
@@ -1,47 +1,57 @@
...
-class ScheduledTaskHandlerTest {
+class ScheduledTaskProcessorTest {
 
     private val clock = Clock.fixed(Instant.now(), ZoneId.systemDefault())
     private val window = 1000L
     private val states = listOf(
-        createStateEntry("key1", clock.instant().minusMillis(window * 2), FlowMapperStateType.CLOSING),
-        createStateEntry("key4", clock.instant().minusMillis(window * 3), FlowMapperStateType.CLOSING)
+        createStateEntry("key1", clock.instant().minusMillis(window * 2)),
+        createStateEntry("key4", clock.instant().minusMillis(window * 3))
```
Comment (emilybowe) [inline components/flow/flow-mapper-service/src/test/kotlin/net/corda/session/mapper/service/executor/ScheduledTaskProcessorTest.kt:33]: key2 might be better here

File: components/flow/flow-mapper-service/src/test/kotlin/net/corda/session/mapper/service/executor/ScheduledTaskProcessorTest.kt [hunk 4]
```diff
@@ -1,47 +1,57 @@
...
     ).toMap()
+    private val inputEvent = Record(
+        Schemas.ScheduledTask.SCHEDULED_TASK_TOPIC_MAPPER_PROCESSOR,
```
Comment (emilybowe) [inline components/flow/flow-mapper-service/src/test/kotlin/net/corda/session/mapper/service/executor/ScheduledTaskProcessorTest.kt:36]: could import Schemas

File: libs/sandbox-internal/src/integrationTest/kotlin/net/corda/sandboxtests/SandboxBundleDifferentLibrariesTest.kt (hunk 2 above)
Comment (JamesHR3) [reply libs/sandbox-internal/src/integrationTest/kotlin/net/corda/sandboxtests/SandboxBundleDifferentLibrariesTest.kt:40]: Yes, I had a failing run because these tests failed with a timeout on receiving these services.

File: charts/corda/values.schema.json [hunk 5]
```diff
@@ -3438,6 +3385,66 @@
...
+                "validationTimeoutSeconds"
+            ],
+            "additionalProperties": false,
+            "properties": {
+                "maxSize": {
+                    "type": "integer",
+                    "default": 5,
+                    "title": "JDBC connection pool size for State Manager DB",
+                    "examples": [
+                        5
```
Comment (emilybowe) [inline charts/corda/values.schema.json:3407]: 5 is a default and an example?

**Review by ronanbrowne — APPROVED** [review]
LGTM

**Review by emilybowe — APPROVED** [review]
LGTM
